nzedb/processing/tv/TMDB.php

```diff
@@ -0,0 +1,347 @@
+							continue;
+						}
+
+						// Download all episodes if new show to reduce API usage
+						if ($this->countEpsByVideoID($videoId) === false) {
+							$this->getEpisodeInfo($tmdbid, -1, -1, '', $videoId);
+						}
+
+						// Check if we have the episode for this video ID
+						$episode = $this->getBySeasonEp($videoId, $seasonNo, $episodeNo, $release['airdate']);
+
+						if ($episode === false) {
+							// Send the request for the episode to TMDB
+							$tmdbEpisode = $this->getEpisodeInfo(
+									$tmdbid,
+									$seasonNo,
+									$episodeNo,
+									$release['airdate']
+							);
+
+							if ($tmdbEpisode) {
+								$episode = $this->addEpisode($videoId, $tmdbEpisode);
+							}
+						}
+
+						if ($episode !== false && is_numeric($episode) && $episode > 0) {
+							// Mark the releases video and episode IDs
+							$this->setVideoIdFound($videoId, $row['id'], $episode);
+							if ($this->echooutput) {
+								echo $this->pdo->log->primary("Found TMDB Match!");
+							}
+							continue;
+						}
+					}
+				} //Processing failed, set the episode ID to the next processing group
+				$this->setVideoNotFound(parent::PROCESS_TRAKT, $row['id']);
+			}
+		}
+	}
+
+	/**
+	 * Calls the API to perform initial show name match to TMDB title
+	 * Returns a formatted array of show data or false if no match
+	 *
+	 * @param $cleanName
+	 *
+	 * @return array|bool
+	 */
+	protected function getShowInfo($cleanName)
+	{
+		$return = $response = false;
+
+		$response = $this->client->searchTVShow($cleanName);
+
+		sleep(1);
+
+		if (is_array($response) && !empty($response)) {
+			$return = $this->matchShowInfo($response, $cleanName);
+		}
+		return $return;
+	}
+
+	/**
+	 * @param $showArr
+	 * @param $cleanName
+	 *
+	 * @return array|bool
+	 */
+	private function matchShowInfo($showArr, $cleanName)
+	{
+		$return = false;
+		$highestMatch = 0;
+
+		foreach ($showArr AS $show) {
+			if ($this->checkRequired($show->_data, 'tmdbS')) {
+				// Check for exact title match first and then terminate if found
+				if (strtolower($show->_data['name']) === strtolower($cleanName)) {
+					$highest = $show;
+					break;
+				} else {
+					// Check each show title for similarity and then find the highest similar value
+					$matchPercent = $this->checkMatch(strtolower($show->_data['name']), strtolower($cleanName), self::MATCH_PROBABILITY);
+
+					// If new match has a higher percentage, set as new matched title
+					if ($matchPercent > $highestMatch) {
+						$highestMatch = $matchPercent;
+						$highest = $show;
+					}
+				}
+			}
+		}
+		if (isset($highest)) {
+			$showAppends = $this->client->getTVShow($highest->_data['id'], 'append_to_response=alternative_titles,external_ids');
+			if ($showAppends) {
+				foreach ($showAppends->_data['alternative_titles']['results'] AS $aka) {
+					$highest->_data['alternative_titles'][] = $aka['title'];
+				}
+				$highest->_data['network'] = $showAppends->_data['networks'][0]['name'];
```

Comment:
I looked at that specifically and was pretty sure I added an isset for network.  TMDB is kind of a PITA, because to get the information we need you have to make two queries.